### DISCUSSION TITLE
cnx-ingress ruules error

### DISCUSSION THREAD

**Discussion by michallawson** [HCL-TECH-SOFTWARE/connections-automation#102] [Q&A] [answered]
When I try to install the component pack I receive the following error:

TASK [component-pack : Setup cnx-ingress Ingress Rule Only] ***************************************************************************************************************************************************
fatal: [cmzr.campbellreith.net]: FAILED! => {"changed": true, "cmd": ["helm", "upgrade", "cnx-ingress-rule", "/opt/hcl-cnx-component-pack/microservices_connections/hybridcloud/helmbuilds/cnx-ingress-0.1.0-20210119-124952.tgz", "-i", "-f", "/home/sysadmin/generated_charts/communityingress.yml", "--namespace", "connections"], "delta": "0:00:01.888636", "end": "2021-04-06 16:18:00.503336", "msg": "non-zero return code", "rc": 1, "start": "2021-04-06 16:17:58.614700", "stderr": "Error: Internal error occurred: failed calling webhook \"validate.nginx.ingress.kubernetes.io\": Post https://cnx-ingress-ingress-nginx-controller-admission.connections.svc:443/networking/v1beta1/ingresses?timeout=10s: dial tcp 10.109.255.47:443: connect: connection refused", "stderr_lines": ["Error: Internal error occurred: failed calling webhook \"validate.nginx.ingress.kubernetes.io\": Post https://cnx-ingress-ingress-nginx-controller-admission.connections.svc:443/networking/v1beta1/ingresses?timeout=10s: dial tcp 10.109.255.47:443: connect: connection refused"], "stdout": "Release \"cnx-ingress-rule\" does not exist. Installing it now.", "stdout_lines": ["Release \"cnx-ingress-rule\" does not exist. Installing it now."]}

If I log onto the server and run the command manually it seems to work as I receive:

[sysadmin@cmzr ~]$ helm upgrade cnx-ingress-rule /opt/hcl-cnx-component-pack/microservices_connections/hybridcloud/helmbuilds/cnx-ingress-0.1.0-20210119-124952.tgz -i -f /home/sysadmin/generated_charts/communityingress.yml --namespace connections

Release "cnx-ingress-rule" has been upgraded. Happy Helming!
NAME: cnx-ingress-rule
LAST DEPLOYED: Tue Apr  6 16:21:41 2021
NAMESPACE: connections
STATUS: deployed
REVISION: 2
TEST SUITE: None

If I continue to run the ansible scripts it then fails when trying to enable the GK flags.

Will be grateful if you can provide any advice you can offer me on how to correct the error.

**Comment by dejo1307** [comment, marked as answer]
Hey Michal,

I saw that started recently in this part here: 

https://github.com/HCL-TECH-SOFTWARE/connections-automation/blob/c9297a53dab06ea1d822952c4605419251f453f1/roles/hcl/component-pack/tasks/setup_community_ingress.yml#L171-L194

What happens is that it tries to apply ingress rules before controller itself is ready to serve the requests (hence it works when you try again manually because more time has passed between). This is the timeout where it is controlled: 

https://github.com/HCL-TECH-SOFTWARE/connections-automation/blob/c9297a53dab06ea1d822952c4605419251f453f1/roles/hcl/component-pack/tasks/setup_community_ingress.yml#L171-L173

I didn't see it happening last two weeks since this was added, but let me test it again.

Regarding GK flags, it is just enabling new community creation wizard. Where exactly/how it is failing?

**Reply by michallawson** [reply]
Thank you and will raise another question as it seems the two may not be related.

Regards

Michael

**Reply by michallawson** [reply]
I increased the period of the pause to 30s and the problem has been resolved.

Thank you for your help.
